packages/contractkit/src/wrappers/MetaTransactionWallet.ts

```diff
@@ -0,0 +1,292 @@
+    return this.executeMetaTransaction(tx, signature)
+  }
+
+  private getMetaTransactionDigestParams = (
+    tx: TransactionInput<any>,
+    nonce: number
+  ): [string, string, string, number] => {
+    const rawTx = toRawTransaction(tx)
+    return [rawTx.destination, rawTx.value, rawTx.data, nonce]
+  }
+
+  getMetaTransactionDigest = proxyCall(
+    this.contract.methods.getMetaTransactionDigest,
+    this.getMetaTransactionDigestParams,
+    stringIdentity
+  )
+
+  private getMetaTransactionSignerParams = (
+    tx: TransactionInput<any>,
+    nonce: number,
+    signature: Signature
+  ): [string, string, string, number, number, string, string] => {
+    const rawTx = toRawTransaction(tx)
+    return [
+      rawTx.destination,
+      rawTx.value,
+      rawTx.data,
+      nonce,
+      signature.v,
+      signature.r,
+      signature.s,
+    ]
+  }
+  getMetaTransactionSigner = proxyCall(
+    this.contract.methods.getMetaTransactionSigner,
+    this.getMetaTransactionSignerParams,
+    stringIdentity
+  )
+
+  eip712DomainSeparator = proxyCall(this.contract.methods.eip712DomainSeparator)
+  isOwner = proxyCall(this.contract.methods.isOwner)
+  nonce = proxyCall(this.contract.methods.nonce, undefined, valueToInt)
+  private getSigner = proxyCall(this.contract.methods.signer, undefined, stringIdentity)
+
+  transferOwnership: (newOwner: Address) => CeloTransactionObject<void> = proxySend(
+    this.kit,
+    this.contract.methods.transferOwnership
+  )
+
+  setSigner: (newSigner: Address) => CeloTransactionObject<void> = proxySend(
+    this.kit,
+    this.contract.methods.setSigner
+  )
+
+  setEip712DomainSeparator: () => CeloTransactionObject<void> = proxySend(
+    this.kit,
+    this.contract.methods.setEip712DomainSeparator
+  )
+
+  /**
+   * Get an cache the chain ID -- assume it's static for a kit instance
+   * @returns chainId
+   */
+  _chainId?: number
+  private async chainId(): Promise<number> {
+    if (this._chainId === undefined) {
+      this._chainId = await this.kit.web3.eth.net.getId()
+    }
+    return this._chainId
+  }
+
+  /**
+   * Get an cache the signer - it should be static for a Wallet instance
+   * @returns signer
+   */
+  _signer?: Address
+  public async signer() {
+    if (this._signer === undefined) {
+      this._signer = await this.getSigner()
+    }
+    return this._signer
+  }
+}
+
+/**
+ * Turns any possible way to pass in a transaction into the raw values
+ * that are actually required. This is used both internally to normalize
+ * ways in which transactions are passed in but also public in order
+ * for one instance of ContractKit to serialize a meta transaction to
+ * send over the wire and be consumed somewhere else.
+ * @param tx TransactionInput<any> union of all the ways we expect transactions
+ * @returns a RawTransactions that's serializable
+ */
+export const toRawTransaction = (tx: TransactionInput<any>): RawTransaction => {
```

Comment:
The way you have the conditional setup will work but I prefer to use `instanceof` when possible so it's explicit that we are routing based on type.

More info on `instanceof` if you need it: https://www.ibrahima-ndaw.com/blog/advanced-typescript-cheat-sheet/#type-guards

Reply:
I agree in principle but in this situation all of the types in the union are interfaces, not classes so `instanceof` can't be used here. The link you shared also has a section on discriminating between interfaces which uses the `in` operator.